Src/unstable/library/network/openssl/src/crypto/asymetric/algoritms/ssl_rsa.e

```diff
@@ -202,11 +250,123 @@ feature -- Sign
 				if l_res <= 0 then
 					create last_error.make (({SSL_CRYPTO_EXTERNALS}.c_error_get_error))
 				else
-					create Result.make_from_string ((create {BASE64}).encoded_string (c_encmsg.string))
+						-- Build an string from the managed pointer
+					create l_data.make (l_count)
+					across c_encmsg.managed_data.read_array (0, l_count) as it loop
+						l_data.append_character (it.item.to_character_8)
+					end
+                    Result := (create {BASE64}).encoded_string (l_data)
+
+				   		-- Base64 encoding using OpenSSL http://doctrina.org/Base64-With-OpenSSL-C-API.html
+				   	-- l_base64 := {SSL_CRYPTO_EXTERNALS}.c_base64_encode (c_encmsg.item, l_count)
+				   	-- create l_result.make_by_pointer (l_base64)
+				    -- Result := l_result.string
 				end
 			end
 
 			{SSL_CRYPTO_EXTERNALS}.c_evp_md_ctx_free (l_sign_ctx)
 		end
 
+	verify_implementation (a_pub_key: SSL_RSA_PUBLIC_KEY; a_plain_text:READABLE_STRING_GENERAL; a_signature: READABLE_STRING_8; a_alg: POINTER): BOOLEAN
+		local
+			l_decoded_sig: STRING_8
+			l_pubkey: POINTER
+				-- EVP_KEY
+			l_res: INTEGER
+			l_verify_ctx: POINTER
+				-- EVP_MD_CTX
+			l_msg: C_STRING
+			l_origin: C_STRING
+			l_count : INTEGER
+			l_padding : INTEGER
+
+			l_b64message: C_STRING
+			l_buffer: C_STRING
+			l_ret: INTEGER
+			l_character: CHARACTER_8
+		do
+			Result := True
+
+			l_count := a_signature.count
+					-- last two chars are =
+			if a_signature.at (l_count) = '=' and then a_signature.at (l_count - 1) = '=' then
+				l_padding := 2
+			else if a_signature.at (l_count) = '=' then
+				l_padding := 1
+			end
+				l_count := (l_count * 3) // 4 - l_padding
+			end
+
+			l_decoded_sig :=(create {BASE64}).decoded_string (a_signature)
+			l_pubkey := {SSL_CRYPTO_EXTERNALS}.c_evp_pkey_new
+			l_res := {SSL_CRYPTO_EXTERNALS}.c_evp_pkey_assign_rsa (l_pubkey, a_pub_key.rsa)
+			if l_res = 0 then
+				create last_error.make (({SSL_CRYPTO_EXTERNALS}.c_error_get_error))
+				Result := False
+			end
+
+			l_verify_ctx:= {SSL_CRYPTO_EXTERNALS}.c_evp_md_ctx_new
+			l_res := {SSL_CRYPTO_EXTERNALS}.c_evp_digestinit_ex (l_verify_ctx, a_alg, default_pointer)
+			if l_res /= 1 then
```

Comment:
should it be  `l_res <= 0`? 

(or change the other  to be `l_res /= 1`)

Reply:
I see, just to be uniform with the rest of the code.